### PR TITLE
Caching the program endpoints

### PR DESCRIPTION
**Pull request by clintonb**
These endpoints are quite slow. Until they have been optimized, the responses must be cached.

ECOM-5709

### CODE REVIEW

**Comment by clintonb** [comment]
@edx/ecommerce please review

File: course_discovery/settings/base.py [hunk 1]
```diff
@@ -331,6 +331,11 @@
     },
 }
 
+REST_FRAMEWORK_EXTENSIONS = {
+    'DEFAULT_CACHE_ERRORS': False,
+    'DEFAULT_CACHE_RESPONSE_TIMEOUT': 1800,
```
Comment (MatthewPiatetsky) [inline course_discovery/settings/base.py:336]: why 1800?

Comment (clintonb) [reply course_discovery/settings/base.py:336]: 30 minutes. Enough time to ensure we don't kill the service, but not so much as to upset our users wanting to make frequent changes.

**Comment by rlucioni** [comment]
👍 , though it's worth keeping in mind that with this caching in place, anyone editing a program will have to wait 30 minutes at a time to see their changes applied.

**Comment by schenedx** [comment]
@clintonb Does the cache TTL have to be 30min? What's the reason behind this number?

**Pull request closed without merging**